frontends/p4/removeDuplicatedSelectLabels.cpp

```diff
@@ -0,0 +1,101 @@
+#include "removeDuplicatedSelectLabels.h"
+
+namespace P4 {
+
+const IR::Node* DoRemoveDuplicatedSelectLabels::postorder(IR::SelectExpression* selectExpression) {
+    auto vecCases = selectExpression->selectCases;
+    std::vector<const IR::Constant*> vecOfConstLabels;
+    std::vector<const IR::Range*> vecOfRangeLabels;
+    // Vector that will contain optimized select casses
+    IR::Vector<IR::SelectCase> newVecCasses;
+
+    for (auto currentExpression : vecCases) {
```

Comment:
This is not really an expression, it's a select case, so the variable name is not the most appropriate.